### ISSUE TITLE
Glossary: Some numbers are letters for some frames

### ISSUE THREAD

**Issue by tolik518** [evanbowman/skyland-beta#11]
I'm not sure if this is fixable at all and the issue is very minor. 
When you switch through the items in the glossary some numbers (esp energy) are shown as capital letters for some frames.

**Comment by evanbowman** [comment]
I fixed it and will include the fixes in the next release.

**Comment by evanbowman** [comment]
Sure! It's kind of a long story :)

A few years ago, when I was learning to program gba games, I made a simple game called BlindJump. Someone wanted to translate the game into another language, so I built unicode support into my text processing engine, and I had to start copying glyphs into video ram dynamically, mainly because I had no way of fitting all of the required chinese words into vram. I ended up implementing a simple garbage collection scheme, where glyphs in vram would be reference counted, and de-allocated from vram when no longer needed.
Fast-forward to 2021. I decided to enter a gba game jam, with only about 20 days remaining. I didn't have time to start from scratch, so I forked the legacy code I'd written for BlindJump, and used it as a starting point for developing Skyland. I decided to add a convenience function for clearing all overlay text from the visible region of the screen, which the glossary implementation calls when switching pages. But this results in all glyph reference counts dropping to zero, so new glyphs allocated in the same frame are created in the same vram slots. What you see happening in the game is essentially tearing.

Rather than fix the text allocator, I've created a lazy band-aid solution for the glossary, by drawing numbers to an offscreen y coordinate, resulting in the vram glyph garbage collector preserving vram for numeric glyphs when clearing and redrawing the screen. https://github.com/evanbowman/skyland-beta/commit/2952dea82eca82e2734242f8af1613471cbfd43e?diff=unified#diff-c690f1cf42171ec5c1bb7263c9cab501ede56a4a128b4c1b2ac53424fbebe434R166

Eventually, I'll find the time to actually fix the text engine
